code/datums/storage/storage.dm

```diff
@@ -22,10 +31,15 @@
 	/// List of all the mobs currently viewing the contents of this storage.
 	VAR_PRIVATE/list/mob/is_using = list()
 
-	/// The storage display screen object.
-	VAR_PRIVATE/atom/movable/screen/storage/boxes
-	/// The 'close button' screen object.
-	VAR_PRIVATE/atom/movable/screen/close/closer
+	/// Associated list of all storage cells and corners.
+	/// In order to minimize memory taken, only one storage UI set is generated per UI theme.
+	VAR_PRIVATE/list/storage_sets = list()
+	/// Associated list of all "close storage" buttons.
+	VAR_PRIVATE/list/atom/movable/screen/close/storage_closers = list()
+	/// Associated list that tracks how many mobs are viewing this storage with a certain UI theme. Used to delete unused UI theme objects.
+	VAR_PRIVATE/list/storage_themes = list()
+	/// Associated list that keeps track of what theme the container was open with. Without this, if observer changes their UI with an open container it will not close and will cause a runtime
+	VAR_PRIVATE/list/theme_cache = list()
```

Comment:
Can we make these lazylists? Considering we probably have like, over a thousand storage datums in existence at world creation and probably 0.01% of those are actually showing people their contents at any given moment, it's free (albeit probably minute) savings